src/api/app/controllers/webui/request_controller.rb

```diff
@@ -57,8 +57,8 @@ def add_reviewer
       request.addreview(opts)
     rescue BsRequestPermissionCheck::AddReviewNotPermitted
       flash[:error] = "Not permitted to add a review to '#{params[:number]}'"
-    rescue ActiveRecord::RecordInvalid, APIError => e
-      flash[:error] = "Unable add review to '#{params[:number]}': #{e.message}"
+    rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotFound, APIError => e
+      flash[:error] = "Unable to add review to request with id '#{params[:number]}': #{e.message}"
```

Comment:
With this change, the `rescue` sections handle exceptions from both lines, the `BsRequest.find_by_number!(params[:number])` and the `request.addreview(opts)` one.

I would split the code more:
- moving the `request = BsRequest.find_by_number!(params[:number])` line to the beginning of the method, and rescue there for the `ActiveRecord::RecordNotFound` exception,
- then fill the `opts` hash
- and lastly create a begin block for the `request.addreview(opts)` call, with the rescue sections we had before.

The reason is that we make clearer what exceptions to expect from running each chunk of code.

Like this:
```ruby
  def add_reviewer
    begin
      request = BsRequest.find_by_number!(params[:number])
    rescue ActiveRecord::RecordNotFound => e
      flash[:error] = "Unable to add review to request with id '#{params[:number]}': #{e.message}"
      return
    end
 
    opts = {}
    case params[:review_type]
    when 'review-user'
      opts[:by_user] = params[:review_user]
    when 'review-group'
      opts[:by_group] = params[:review_group]
    when 'review-project'
      opts[:by_project] = params[:review_project]
    when 'review-package'
      opts[:by_project] = params[:review_project]
      opts[:by_package] = params[:review_package]
    end
    opts[:comment] = params[:review_comment] if params[:review_comment]

    begin
      request.addreview(opts)
    rescue BsRequestPermissionCheck::AddReviewNotPermitted
      flash[:error] = "Not permitted to add a review to '#{params[:number]}'"
    rescue ActiveRecord::RecordInvalid, APIError => e
      flash[:error] = "Unable to add review to request with id '#{params[:number]}': #{e.message}"
    end
  end
```

Reply:
I'm not against this suggestion, but I find it out of scope for this PR. I prefer to not do this here.

Reply:
Ok.